[Themes] Fix MuiThemeProvider default theme

mbrookes opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! -->

- [ ] PR has tests / docs demo, and is linted.
- [x] Commit and PR titles begin with [ComponentName], and are in imperative form: "[Component] Fix leaky abstraction".
- [x] Description explains the issue / use-case resolved, and auto-closes the related issue(s) (http://tr.im/vFqem).

 `muiTheme` need not be a required prop, as a default is provided.

Simplify muiThemeable (although I'm not convinced of the value of this module).

Update and reorder the docs.

nathanmarks commented:
@mbrookes I'm pretty sure that we should be able to get rid of `muiThemeable`. Isn't that left over from when we had all that context boilerplate in the components?

mbrookes commented:
I'm not sure why it was needed even then...

mbrookes commented:
Ok, here's the original PR: https://github.com/callemall/material-ui/pull/2493

> @newoga 
>  It allows us to develop material-ui components without having to use context directly, and instead receive the muiTheme as a prop (even though the higher order component is providing the muiTheme prop from context).

Now that we've gone all-in on reading the theme from context, I think it's safe to remove muiTheme.

@newoga - sorry to undo your work, but any objections?

nathanmarks commented:
@mbrookes 

![img](https://avatars0.githubusercontent.com/u/344018?v=3&s=460)

> ## Yeah go for it! no problem!

newoga commented:
LOL

newoga commented:
> @newoga - sorry to undo your work, but any objections?

Yeah this was the plan, since we no longer have default themes. Some components were relying on it such as Divider and Subheader but after the switch it's not the case anymore

mbrookes commented:
It's dead Jim.

mbrookes commented:
I've marked this as a breaking change, although the probability that anyone was using this outside of Material-UI is pretty slim.

nathanmarks commented:
@mbrookes was it even "public" 😄

mbrookes commented:
It was documented.

newoga commented:
If we're going to start breaking changes/removing things, might as well remove theme decorator and theme manager too

mbrookes commented:
That's a good point. Let's hold that thought for the first 0.16.0 pre, and I'll just fix the regression in this one.

mbrookes commented:
I put it back, but I've left it out of the docs so as to discourage its use.

oliviertassinari commented on docs/src/app/components/pages/get-started/usage.md:
```diff
@@ -13,7 +13,7 @@ import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
 import MyAwesomeReactComponent from './MyAwesomeReactComponent';
 
 const App = () => (
-  <MuiThemeProvider muiTheme={getMuiTheme()}>
```
We can remove
```js
import getMuiTheme from 'material-ui/styles/getMuiTheme';
```.

mbrookes commented:
@oliviertassinari Done.

oliviertassinari commented on docs/src/app/components/pages/customization/themes.md:
```diff
@@ -1,11 +1,41 @@
+### Predefined themes
+
+We ship two base themes with Material-UI: light and dark. They are located
+under [`material-ui/lib/styles/baseThemes/`](https://github.com/callemall/material-ui/blob/master/src/styles/baseThemes/).
```
`material-ui/lib/styles/baseThemes/` doesn't match with `master/src/styles/baseThemes/`.

mbrookes commented:
I've updated the themes doc and added a note to lightBaseTheme to keep the docs up to date if it changes. But I noticed this: `clockCircleColor: fade(darkBlack, 0.07),` which looks more like it should belong in `getMuiTheme`... not one to fix here, but thought I'd mention it while I saw it.

mbrookes commented:
Should be good to go.

oliviertassinari commented:
@mbrookes That looks good 😄.
